solver/src/interactions/weth.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{encoding::EncodedInteraction, settlement::Interaction};
+use contracts::WETH9;
+use primitive_types::U256;
+
+#[derive(Debug)]
+pub struct UnwrapWethInteraction {
+    pub weth: WETH9,
+    pub amount: U256,
+}
+
+impl Interaction for UnwrapWethInteraction {
+    fn encode(&self) -> Vec<EncodedInteraction> {
+        let method = self.weth.withdraw(self.amount);
+        let calldata = method.tx.data.expect("no calldata").0;
+        vec![(self.weth.address(), 0.into(), calldata)]
```

Comment:
Unrelated to this PR, but wondering if we should refactor EncodedInteraction to a named struct to make this read a bit nicer?

Reply:
One issue here is that `EncodedInteraction` represents the type that is generated by `ethcontract` (because it doesn't support ABI named tuples).

I think the best way to achieve this would actually be to add support for named tuples in `ethcontract` instead of adding an intermediate `InteractionData` representation (although the latter may be a nice stepping stone for now).